src/equations/shallow_water_two_layer_1d.jl

```diff
@@ -0,0 +1,536 @@
+  - g\frac{\partial}{\partial x} \left(h_2 + h_1 + h_0 \right)
+  \end{aligned}
+  ```
+  The unknown quantities of the system are the water heights of the lower layer ``h_2`` and the 
+  upper layer ``h_1`` with respecitv velocities ``v_1`` and ``v_2``. The gravitational constant is 
+  denoted by `g`, the layer densitites by ``\rho_1``and ``\rho_2`` and the (possibly) variable 
+  bottom topography function ``b(x)``. The conservative variable water height ``h_2`` is measured 
+  from the bottom topography ``b`` and ``h_1`` relative to ``h_2``, therefore one also defines the 
+  total water heights as ``H1 = h1 + h2 + b`` and ``H2 = h2 + b``.
+
+  The additional quantity ``H_0`` is also available to store a reference value for the total water
+  height that is useful to set initial conditions or test the "lake-at-rest" well-balancedness.
+
+  The bottom topography function ``b(x)`` is set inside the initial condition routine
+  for a particular problem setup. To test the conservative form of the SWE one can set the bottom 
+  topography variable `b` to zero.
+
+  In addition to the unknowns, Trixi currently stores the bottom topography values at the 
+  approximation points despite being fixed in time. This is done for convenience of computing the 
+  bottom topography gradients on the fly during the approximation as well as computing auxiliary 
+  quantities like the total water height ``H`` or the entropy variables.
+  This affects the implementation and use of these equations in various ways:
+  * The flux values corresponding to the bottom topography must be zero.
+  * The bottom topography values must be included when defining initial conditions, boundary 
+    conditions or source terms.
+  * [`AnalysisCallback`](@ref) analyzes this variable.
+  * Trixi's visualization tools will visualize the bottom topography by default.
+
+  A good introduction for the two-layer SWE is available in Chapter 12 of the book:
+  - Benoit Cushman-Roisin (2011)
+    Introduction to geophyiscal fluid dynamics: physical and numerical aspects
+  """
+  struct TwoLayerShallowWaterEquations1D{RealT<:Real} <: AbstractShallowWaterEquations{1,5}
+    gravity::RealT # gravitational constant
+    H0::RealT      # constant "lake-at-rest" total water height
+    rho1::RealT    # lower layer density
+    rho2::RealT    # upper layer density
+  end
+
+  # Allow for flexibility to set the gravitational constant within an elixir depending on the
+  # application where `gravity_constant=1.0` or `gravity_constant=9.81` are common values.
+  # The reference total water height H0 defaults to 0.0 but is used for the "lake-at-rest"
+  # well-balancedness test cases. Densities must be specificed such that rho_1 < rho_2.
+  function TwoLayerShallowWaterEquations1D(; gravity_constant, H0=0.0, rho1, rho2)
+    TwoLayerShallowWaterEquations1D(gravity_constant, H0, rho1, rho2)
+  end
+
+  have_nonconservative_terms(::TwoLayerShallowWaterEquations1D) = Val(true)
```

Comment:
```suggestion
  have_nonconservative_terms(::TwoLayerShallowWaterEquations1D) = True()
```

Reply:
Done in commit 12e1e0058e4c787239293786ad641aa71d6a6444